include/dali/c_api.h

```diff
@@ -309,6 +309,33 @@ DLL_PUBLIC void daliDeletePipeline(daliPipelineHandle *pipe_handle);
  */
 DLL_PUBLIC void daliLoadLibrary(const char *lib_path);
 
+  /**
+   * @brief Returns the epoch size with/without padded samples for given, named reader
+   */
+  DLL_PUBLIC int64_t daliGetEpochSize(daliPipelineHandle* pipe_handle, const char *reader_name,
+                                  bool with_padding);
+
+  /**
+   * @brief Returns the shard ID for given, named reader
+   */
+  DLL_PUBLIC int daliGetShardId(daliPipelineHandle* pipe_handle, const char *reader_name);
+
+  /**
+   * @brief Returns the number of shards, named reader
+   */
+  DLL_PUBLIC int daliGetNumberOfShards(daliPipelineHandle* pipe_handle, const char *reader_name);
+
+  /**
+   * @brief Returns if reader pads last batch for, named reader
+   */
+  DLL_PUBLIC bool daliIsPadLastBatch(daliPipelineHandle* pipe_handle, const char *reader_name);
+
+  /**
+   * @brief Returns if reader sticks to the shard, named reader
```

Comment:
```suggestion
   * @brief Returns if reader sticks to the shard for given named reader
```
Was this the intention?

Reply:
Done